### PR TITLE
[release][no_ci] remove datasets_ingest_train_infer and adjust datasets_preprocess_ingest.

### PR DESCRIPTION
**Pull request by xwjiang2010**
remove datasets_ingest_train_infer and adjust datasets_preprocess_ingest.

Signed-off-by: xwjiang2010 <xwjiang2010@gmail.com>

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/ray_sgd_training_compute_no_gpu.yaml [hunk 1]
```diff
@@ -20,8 +20,8 @@ head_node_type:
 worker_node_types:
     - name: memory_node
       instance_type: i3.8xlarge
-      min_workers: 20
```
Comment (xwjiang2010) [inline release/nightly_tests/dataset/ray_sgd_training_compute_no_gpu.yaml:23]: @rkooo567 Sang, do you remember what may be the reason that we request 20 worker nodes when the test only needs 16?

**Comment by xwjiang2010** [comment]
https://buildkite.com/ray-project/release-tests-pr/builds/18508 

datasets_preprocess_ingest test is passing.

**Review by clarkzinzow — APPROVED** [review]
LGTM! Can we also delete the `ray_sgd_training` Python script, cluster env yaml, cluster compute yaml, etc., or are those being used for something else as well?

File: release/nightly_tests/dataset/ray_sgd_training_compute_no_gpu.yaml (hunk 1 above)
Comment (rkooo567) [reply release/nightly_tests/dataset/ray_sgd_training_compute_no_gpu.yaml:23]: Not sure. I don't think I added this test...

**Review by rkooo567 — APPROVED** [review]
Just a stamp... (no context to review the PR technically lol...)

**Pull request merged**